packages/manager/modules/pci/src/projects/project/serving/namespace/models/add/add.html

```diff
@@ -0,0 +1,138 @@
+<div class="servingAddModel">
+    <h1 data-translate="pci_projects_project_serving_namespace_models_add"></h1>
+
+    <oui-stepper data-on-finish="$ctrl.addModel()">
+
+        <oui-step-form
+            data-header="{{ :: 'pci_projects_project_serving_namespace_models_add_type' | translate }}"
+            data-editable="!$ctrl.isAdding"
+            data-loading="$ctrl.isAdding"
+        >
+            <oui-radio
+                data-ng-repeat="kind in $ctrl.types track by $index"
+                data-model="$ctrl.model.kind"
+                data-name="kind"
+                data-value="kind"
+                data-required>
+                <span
+                    data-translate="{{:: 'pci_projects_project_serving_namespace_models_add_type_' + kind.toLowerCase() }}"></span>
+                <button
+                    type="button"
+                    class="oui-popover-button"
+                    data-oui-popover="{{:: 'pci_projects_project_serving_namespace_models_add_type_' + kind.toLowerCase() + '_popover' | translate }}"
+                    data-oui-popover-placement="auto"></button>
+            </oui-radio>
+        </oui-step-form>
+
+        <oui-step-form
+            data-header="{{ :: 'pci_projects_project_serving_namespace_models_add_model' | translate }}"
+            data-editable="!$ctrl.isAdding"
+            data-loading="$ctrl.isAdding">
+            <div class="container-fluid px-0" data-ng-if="$ctrl.model.kind == 'preset'">
+                <div class="row">
+                    <div class="col-md-6 col-xl-4 my-3"
+                         data-ng-repeat="preset in $ctrl.presets track by $index">
```

Comment:
Is there a more accurate tracking that could be used ? Something like an `id` maybe ? 
```suggestion
                         data-ng-repeat="preset in $ctrl.presets track by preset.id">
```

Reply:
fixed 9eb0962dad0527f7e808bd6b11ebe3c3cfec136e